### PR TITLE
FEAT: Add details to README

### PR DESCRIPTION
**Pull request by guzzur**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,114 @@
-# strigo-github-exercises-examples
-Examples for markdown exercises that may be used in Strigo training events
+# Strigo GitHub Class Exercises examples
+
+## Quick start
+
+1. Create `.strigo/config.yml` file in your exercises repository
```
Comment (nir0s) [inline README.md]: (or fork this repo).

File: README.md [hunk 2]
```diff
@@ -1,2 +1,114 @@
...
+1. Configure exercises in `.strigo/config`
```
Comment (nir0s) [inline README.md]: How about: "Edit the `config.yml` file"?

File: README.md [hunk 3]
```diff
@@ -1,2 +1,114 @@
...
+   ```yaml
+   ---
+   classes:
+     - exercises:
+         # `file` is relative to .strigo directory
+         - file: "../Exercise.md"
+           title: "Title!"
+   ```
+
+1. In Strigo Class editor, fetch Exercises for a desired Class
```
Comment (nir0s) [inline README.md]: I'd use "Edit your class template in Strigo and chose to fetch exercises from GitHub".

File: README.md [hunk 4]
```diff
@@ -1,2 +1,118 @@
...
+1. Done!
+
+---
+
+## Config file structure
+
+- `classes` - array of Classes
```
Comment (nir0s) [inline README.md]: In YAML, there's no "array", only "list".

Comment (nir0s) [reply README.md]: Also, array of Classes => A list of classes (required)

File: README.md [hunk 5]
```diff
@@ -1,2 +1,118 @@
...
+  - `class_id` - defines the Class ID, optional if the configuration has only one Class
```
Comment (nir0s) [inline README.md]: I'd do: "The ID of the class you'd like to fetch exercises for (optional if the repo only supports one class)."

File: README.md [hunk 6]
```diff
@@ -1,2 +1,118 @@
...
+  - `exercises` - array of Class' Exercises, cannot be empty
```
Comment (nir0s) [inline README.md]: "A list of class exercises (required)"

File: README.md [hunk 7]
```diff
@@ -1,2 +1,118 @@
...
+    - `file` - path to Exercise markup file, relative to `.strigo` directory, supported extensions `.md` (markdown) and `.rst` (restructured), any other extension should be overriden by `syntax` field (see below)
```
Comment (nir0s) [inline README.md]: "The path to the markup file (relative to the `.strigo` directory). Supported extensions are `.md` (Markdown) and `.rst` (reStructured Text). Any other extension should be overridden by the `syntax` field (see below)."

File: README.md [hunk 8]
```diff
@@ -1,2 +1,118 @@
...
+    - `title` - defines Exercise title in Strigo interface
```
Comment (nir0s) [inline README.md]: "The title of the exercise"

File: README.md [hunk 9]
```diff
@@ -1,2 +1,118 @@
...
+    - `syntax` - overrides `file` extension, supported syntaxes `.md` (markdown) and `.rst` (restructured)
```
Comment (nir0s) [inline README.md]: "Overrides the file extension as used in the `file` field (`.md`/`.rst`)."

File: README.md [hunk 10]
```diff
@@ -1,2 +1,118 @@
...
+
+---
+
+## Main concepts
+
+1. **Hello world**
+
+   Just configure the order of the files, set titles and it is good to go.
```
Comment (nir0s) [inline README.md]: it is => you're

File: README.md [hunk 11]
```diff
@@ -1,2 +1,118 @@
...
+         # `file` is relative to .strigo directory
+         - file: "../hello-world/hello.md"
+           title: "Hello!"
+         - file: "../hello-world/world.txt"
+           title: "World!"
+   ```
+
+1. **Multiple Classes in a single repository**
+
+   You can have multiple Classes in a single repository, defining Class IDs for each.
```
Comment (nir0s) [inline README.md]: have => support
Classes => classes
defining Class IDs for each => by defining class IDs for each class.

File: README.md [hunk 12]
```diff
@@ -1,2 +1,118 @@
...
+           title: "Title 2!"
+     - class_id: "<class_id_2>"
+       exercises:
+         - file: "../3.md"
+           title: "Title 3!"
+         - file: "../4.md"
+           title: "Title 4!"
+   ```
+
+1. **Reusing Exercises for different classes**
```
Comment (nir0s) [inline README.md]: lowercase E

File: README.md [hunk 13]
```diff
@@ -1,2 +1,118 @@
...
+
+   There is no restriction on using the same file for multiple Classes, or even using the same file multiple times for the same Class
+
+   ```yaml
+   ---
+   classes:
```
Comment (nir0s) [inline README.md:78]: I'd consider YAML shortcuts here, like in https://github.com/strigo/taco/blob/master/.circleci/config.yml#L61

File: README.md [hunk 14]
```diff
@@ -1,2 +1,118 @@
...
+           title: "Title 1!"
+     - class_id: "<class_id_2>"
+       exercises:
+         - file: "../1.md"
+           title: "Title 1!"
+   ```
+
+1. **Markup syntaxes**
+
+   You can use different markup syntaxes by using `syntax` field for each Exercise
```
Comment (nir0s) [inline README.md]: ...the `syntax` field...

File: README.md [hunk 15]
```diff
@@ -1,2 +1,118 @@
...
+           # uses file extension as fallback
+           syntax: "md"
+         - file: "../2.txt"
+           title: "Title 1!"
+           syntax: "rst"
+   ```
+
+1. **Relative vs Absolute paths to images**
+
+   Your Exercises' repository can have images stored inside the repository, a.k.a Relative images. Fetching these exercises will upload these images to Strigo. If the image could not be accessed, we will replace it with a placeholder instead.
```
Comment (nir0s) [inline README.md]: "Your exercises repository can have images stored within it. Fetching these exercises will upload the images to Strigo. If the image could be accessed for some reason, we will replace it with a placeholder instead"

"Absolute image paths will not be affected, so if the image cannot be accessed, it won't show up for your students."

File: README.md [hunk 16]
```diff
@@ -1,2 +1,118 @@
...
+
+   Absolute images paths will not be affected, so if the image cannot be accessed, it won't show up for your students.
+
+1. **Public vs Private repository**
```
Comment (nir0s) [inline README.md]: repository => repositories

File: README.md [hunk 17]
```diff
@@ -1,2 +1,118 @@
...
+
+   There is no difference between Public and Private repositories configuration structure. Please follow [Strigo Guidelines](http://help.strigo.io/en/articles/4951906-fetching-lab-exercises-from-github) for more details.
```
Comment (nir0s) [inline README.md]: "Both public and private repositories should be configured the same way." ...

"Strigo's Guidelines"

**Review by nir0s — CHANGES_REQUESTED** [review]

File: README.md (hunk 3 above)
Comment (guzzur) [reply README.md]: We don't use the term "template" in Lounge, so maybe just "Edit your class"?

File: README.md (hunk 4 above)
Comment (guzzur) [reply README.md]: They actually call it a "sequence" but there are lots of "array" mentions as well. I can live with a "list" :)

File: README.md (hunk 5 above)
Comment (guzzur) [reply README.md]: "supports" sounds wrong to me. "contains"?

File: README.md (hunk 11 above)
Comment (guzzur) [reply README.md]: "support"... not sure it means what we try to mean

File: README.md (hunk 12 above)
Comment (guzzur) [reply README.md]: Huh.. why? It is capitalized in all other cases

File: README.md (hunk 3 above)
Comment (nir0s) [reply README.md]: We use class template when talking to customers all the time.. but I agree that it's inconsistent. "class" is good.

File: README.md (hunk 4 above)
Comment (nir0s) [reply README.md]: Oh.. true.

File: README.md (hunk 5 above)
Comment (nir0s) [reply README.md]: I think "supports" actually makes more sense than contains. The repo doesn't "contain" a class.

File: README.md (hunk 13 above)
Comment (guzzur) [reply README.md:78]: Prefer not to. In this way it is easily copy-pasteable.

File: README.md (hunk 11 above)
Comment (nir0s) [reply README.md]: Hmm. "configure"?

File: README.md (hunk 12 above)
Comment (nir0s) [reply README.md]: yup.. and it doesn't make sense to me. "Exercises" isn't a name. It's a noun.

File: README.md (hunk 13 above)
Comment (nir0s) [reply README.md:78]: Well, it's copy-pastable even if you use shortcuts. You repeat the same two lines (title, file) multiple times.

File: README.md (hunk 12 above)
Comment (guzzur) [reply README.md]: But... we did the opposite in Lounge. What is the difference?

File: README.md (hunk 13 above)
Comment (guzzur) [reply README.md:78]: That's exactly the point of this example, it shows that you can reuse the same file many times. I prefer not to make it too complicated, I am assuming that not everyone is familiar with YAML, and think this should be done in a simplest possible way.

File: README.md (hunk 12 above)
Comment (nir0s) [reply README.md]: It's wrong... then. Can I explain f2f when we talk?

Comment (guzzur) [reply README.md]: Sure. Anyway, done.

File: README.md [hunk 18]
```diff
@@ -1,2 +1,119 @@
-# strigo-github-exercises-examples
-Examples for markdown exercises that may be used in Strigo training events
+# Strigo GitHub class exercises examples
```
Comment (nir0s) [inline README.md:1]: exercises => exercise

File: README.md [hunk 19]
```diff
@@ -1,2 +1,119 @@
...
+
+## Quick start
+
+1. Create `.strigo/config.yml` file in your exercises' repository (or fork this repository).
```
Comment (nir0s) [inline README.md:5]: Create a

exercises' => exercises

File: README.md [hunk 20]
```diff
@@ -1,2 +1,119 @@
...
+       exercises:
+         - file: "../3.md"
+           title: "Title 3!"
+         - file: "../4.md"
+           title: "Title 4!"
+   ```
+
+1. **Reusing exercises for different classes**
+
+   There is no restriction on using the same file for multiple classes, or even using the same file multiple times for the same class
```
Comment (nir0s) [inline README.md:74]: period

File: README.md [hunk 21]
```diff
@@ -1,2 +1,119 @@
...
+           # `syntax` is optional
+           # supported syntaxes are "md" | "rst"
+           # uses file extension as fallback
+           syntax: "md"
+         - file: "../2.txt"
+           title: "Title 1!"
+           syntax: "rst"
+   ```
+
+1. **Relative vs Absolute paths to images**
```
Comment (nir0s) [inline README.md:111]: Absolute => absolute

File: README.md [hunk 22]
```diff
@@ -1,2 +1,119 @@
...
+
+   Your exercises repository can have images stored within it. Fetching these exercises will upload the images to Strigo. If the image could be accessed for some reason, we will replace it with a placeholder instead.
+
+   Absolute image paths will not be affected, so if the image cannot be accessed, it won't show up for your students.
+
+1. **Public vs Private repositories**
```
Comment (nir0s) [inline README.md:117]: Private => private

File: README.md [hunk 23]
```diff
@@ -1,2 +1,119 @@
...
+           # uses file extension as fallback
+           syntax: "md"
+         - file: "../2.txt"
+           title: "Title 1!"
+           syntax: "rst"
+   ```
+
+1. **Relative vs Absolute paths to images**
+
+   Your exercises repository can have images stored within it. Fetching these exercises will upload the images to Strigo. If the image could be accessed for some reason, we will replace it with a placeholder instead.
```
Comment (nir0s) [inline README.md:113]: could => could not

**Review by nir0s — APPROVED** [review]
Approved. Just some small comments.

File: README.md (hunk 23 above)
Comment (guzzur) [reply README.md:113]: oi

**Pull request merged**